aptos-move/move-examples/bonding_curve_launchpad/README.md

```diff
@@ -0,0 +1,109 @@
+   3. A pre-defined number of the FA is minted, and kept on `bonding_curve_launchpad`.
+2. `bonding_curve_launchpad` creates a liquidity pair on the `liquidity_pairs` module, which follows the constant product formula.
+   1. Liquidity pair is represented and stored as an Object. This allows for the reserves to be held directly on the object, rather than the `bonding_curve_launchpad` account.
+   2. The entirety of minted FA + a pre-defined number of **virtual liquidity** for APT is deposited into the liquidity pair object. 
+   3. Trading against the liquidity pair is enabled, but restricted to `public entry` functions found in `bonding_curve_launchpad`.
+3. Optionally, the creator can immediately initiate a swap from APT to the FA.
+#### Trading against the FA's associated liquidity pair
+1. External users can swap APT to FA, or vice versa, through `public entry` methods available on `bonding_curve_launchpad`. 
+   1. Although the normal `transfer` functionality of the FA is disabled by the custom dispatchable `withdraw` function, `bonding_curve_launchpad` can assist with swaps using it's stored `transfer_ref` from the FA `SmartTable`. `transfer_ref` is not impeded by the custom dispatchable function.
+   2. The logic for calculating the `amountOut` of a swap is based on the constant product formula for reader familiarity. In a production scenario, a sub-linear trading function can assist in incentivizing general early adoption.
+#### Graduating from `liquidity_pairs` to a public FA DEX
+1. After each swap from APT to FA, when the APT reserves are increasing, a threshold is checked for whether the liquidity pair can **graduate** or not. The threshold is a pre-defined minimum amount of APT that must exist in the reserves. Once this threshold is met during a swap, **graduation** begins.
+   1. The associated liquidity pair on the `liquidity_pairs` module is disabled by toggling `is_enabled`, preventing any more swaps against the pair. Additionally, the liquidity pair's `is_frozen` is disabled to allow owners to transfer freely. 
+   2. The reserves from the liquidity pair, both APT and FA, are moved to an external, public third-party DEX as a new liquidity pair. In this case, the `aptos-move` FA DEX example, called swap. 
+   3. To prevent any wrongdoing from the `bonding_curve_launchpad` owner, any liquidity tokens received during the creation of the new liquidity pair on the third-party DEX will be sent to a dead address. Otherwise, the tokens could be used to drain the liquidity pair, at any time.
```

Comment:
Just curious, why dead address? The lp tokens should be sent to those who contributed to the pair, to continue earning rewards? Can we not track this in the LP struct?

Reply:
LP tokens should only be given to people providing liquidity. During the `liquidity_pairs` timeline of the FA (before graduation), the users can only swap. They aren't able to provide LP, so they don't have corresponding LP tokens to receive.

There's not much point of determining a "fair" method of distribution for LP tokens generated during graduation, either. Additional value can't and won't be generated from doing this.